src/engine/controls/bpmcontrol.cpp

```diff
@@ -827,22 +834,41 @@ double BpmControl::updateLocalBpm() {
 
 double BpmControl::updateBeatDistance() {
     double beat_distance = getBeatDistance(getSampleOfTrack().current);
-    m_pThisBeatDistance->set(beat_distance);
+    if (m_dUserRateTweak != 0.0) {
+        double master_percentage = m_dSyncTargetBeatDistance.getValue();
```

Comment:
This is only valid if we are a follower, are we? Can we read it from m_dUserRateTweak as well? I think in this case we need to reset it once we are not in the follower state anymore.

Reply:
Ah, yes, right no need to set if we aren't a follwer.  We do reset it if we become master via  calcSyncedRate which requests a resetSyncAdjustment